extensions/pyRevitTools.extension/pyRevit.tab/Modify.panel/3D.pulldown/Section Box Navigator.pushbutton/sectionbox_actions.py

```diff
@@ -0,0 +1,158 @@
+from pyrevit import revit, script, forms
+from pyrevit.framework import List, Math
+from pyrevit.compat import get_elementid_value_func
+from pyrevit import DB, UI
+
+get_elementid_value = get_elementid_value_func()
+
+logger = script.get_logger()
+
+
+def toggle(doc, datafilename):
+    """Toggle section box."""
+    current_view = doc.ActiveView
+    if not isinstance(current_view, DB.View3D):
+        return
+    current_view_id_value = get_elementid_value(current_view.Id)
+    sectionbox_active_state = current_view.IsSectionBoxActive
+
+    try:
+        view_boxes = script.load_data(datafilename)
+    except Exception:
```

Comment:
I'll write it here, but it is valid for all of the try/except blocks: be explicit about the exception to catch! for this context it doesn't matter, but if this was a regular python script, this would block the ability to stop the script with CTRL/CMD+C